src/app/services/task.service.ts

```diff
@@ -1,54 +1,55 @@
 })
 export class TaskService {
-    public tasks$: Subject<TaskType[]> = new Subject();
-    private tasks: TaskType[] = [];
-
-    constructor() {
-    }
-
-    addTask(title: string): void {
-        this.tasks.push({
-            title: title,
-            complete: false
-        });
-        this.tasks$.next(this.tasks);
-    }
-
-    getTasks(filter: string = 'all'): TaskType[] {
-        if (filter === 'active') {
-            return this.tasks.filter((item: TaskType) => !item.complete);
-        }
-
-        if (filter === 'completed') {
-            return this.tasks.filter((item: TaskType) => item.complete);
-        }
-
-        return this.tasks;
-    }
-
-    removeTask(task: TaskType): void {
-        this.tasks = this.tasks.filter((item: TaskType) => item !== task);
-        this.tasks$.next(this.tasks);
-    }
-
-    clearCompleted(filterValue: string): void {
-        this.tasks = this.tasks.filter((item: TaskType) => !item.complete);
-        this.tasks$.next(this.getTasks(filterValue));
-    }
-
-    toggleAllTasks(): boolean {
-        if (this.tasks.some((item: TaskType) => !item.complete)) {
-            this.tasks.map(item => item.complete = true);
-            return true;
-        } else {
-            this.tasks.map(item => item.complete = false);
-            return  false;
-        }
-    }
+  public tasks$: BehaviorSubject<TaskType[]> = new BehaviorSubject<TaskType[]>([]);
+  public countLeft$: BehaviorSubject<number> = new BehaviorSubject<number>(0);
+  public isSomeCompleted$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+
+  addTask(title: string): void {
+    const oldTasks: TaskType[] = this.tasks$.getValue();
+    const newTask: TaskType = {
+      id: Date.now(),
+      title: title,
+      complete: false
+    };
+    this.tasks$.next([...oldTasks, newTask]);
+  }
+
+  getTasks(filters: string = 'all'): BehaviorSubject<TaskType[]> {
+    let tasks = this.tasks$.getValue();
+    tasks = filters === 'all' ? tasks : tasks.filter((item: TaskType) => filters === 'active' ? !item.complete : item.complete);
+    return new BehaviorSubject<TaskType[]>(tasks);
+  }
+
+  removeTask(task: TaskType): void {
+    const oldTasks: TaskType[] = this.tasks$.getValue();
+    const oldTasksIndex: number = oldTasks.findIndex((item: TaskType) => item.id === task.id);
+    oldTasks.splice(oldTasksIndex, 1);
+    this.tasks$.next(oldTasks);
+  }
+
+  clearCompleted(): void {
+    const oldTasks: TaskType[] = this.tasks$.getValue();
+    const newTasks: TaskType[] = oldTasks.filter((item: TaskType) => !item.complete);
+    this.tasks$.next(newTasks);
+  }
+
+  toggleAllTasks(): void {
+    const oldTasks: TaskType[] = this.tasks$.getValue();
+    const someCompleted = oldTasks.some((item: TaskType) => !item.complete);
+    oldTasks.map(item => item.complete = someCompleted);
+  }
+
+  getCountLeftTasks(): void {
+    this.countLeft$.next(this.tasks$.getValue().filter((item: TaskType) => !item.complete).length);
```

Comment:
чаще пользуйся цепочкой операторов из rxjs
```suggestion
    return this.tasks$.tap(map(items => items.length))
```